### PR TITLE
Add description about PoA/PoS switch

### PR DESCRIPTION
**Pull request by Kourin1996**
Fixes EDGE-63

This PR adds description about PoA/PoS switch function, which will be added by https://github.com/0xPolygon/polygon-edge/pull/334

### CODE REVIEW

File: docs/consensus/migration-to-pos.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+id: migration-to-pos
+title: Migration from PoA to PoS
+---
+
+## Overview
+
+This section guides you through migration from PoA to PoS for the running cluster without reset of blockchain.
```
Comment (zivkovicmilos) [inline docs/consensus/migration-to-pos.md:8]: I would just rephrase the last section of this sentence:

`This section guides you through the migration from PoA to PoS IBFT modes, and vice versa, for a running cluster - without the need to reset the blockchain.`

File: docs/consensus/migration-to-pos.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## How to migrate to PoS
+
+You will need to stop all nodes, update genesis.json, and restart the nodes in order to switch to PoS. `ibft switch` command append fork configuration into `genesis.json`.
```
Comment (zivkovicmilos) [inline docs/consensus/migration-to-pos.md:12]: What's unclear to me - by reading this I get the impression that I need to stop the nodes, _manually edit the genesis.json_ and restart the nodes.

If I understood correctly, `ibft switch` does it for us while the cluster is running, it's just that we need to restart the nodes after we run `ibft switch`?

File: docs/consensus/migration-to-pos.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+````bash
+polygon-edge ibft switch --chain ./genesis.json --type PoS --deployment 100 --from 200
+````
+
+To switch to PoS, you will need to specify 2 block heights: `deployment` and `from`. `deployment` is the height to deploy the staking contract and `from` is the height of beginning of PoS. The staking contract will be deployed at the address `0x0000000000000000000000000000000000001001`  at the `deployment`, like as the case of pre-deployed contract.
```
Comment (zivkovicmilos) [inline docs/consensus/migration-to-pos.md:18]: Please add links to the Staking SC doc / repo here 🙏

File: docs/consensus/migration-to-pos.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+Each validator needs to stake after contract is deployed at `deployment` and before `from` in order to be a validator at the beginning of PoS. Each validator will update own validator set by the set in the staking contract at the beginning of PoS.
```
Comment (zivkovicmilos) [inline docs/consensus/migration-to-pos.md:20]: This should be inside a :::warning tag

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I've left some minor improvement suggestions 🙏

File: docs/consensus/migration-to-pos.md (hunk 2 above)
Comment (Kourin1996) [reply docs/consensus/migration-to-pos.md:12]: yes

File: docs/consensus/migration-to-pos.md (hunk 4 above)
Comment (Kourin1996) [reply docs/consensus/migration-to-pos.md:23]: Fixed at c54b7deb2c83464f83a348a110ea278cef3f0798

File: docs/consensus/migration-to-pos.md (hunk 3 above)
Comment (Kourin1996) [reply docs/consensus/migration-to-pos.md:18]: Fixed at c54b7deb2c83464f83a348a110ea278cef3f0798

File: docs/consensus/migration-to-pos.md (hunk 2 above)
Comment (Kourin1996) [reply docs/consensus/migration-to-pos.md]: Fixed at c54b7deb2c83464f83a348a110ea278cef3f0798

File: docs/consensus/migration-to-pos.md (hunk 1 above)
Comment (Kourin1996) [reply docs/consensus/migration-to-pos.md]: Fixed at c54b7deb2c83464f83a348a110ea278cef3f0798

**Review by Aleksao998 — APPROVED** [review]
Thank you for this PR. Well done!

**Review by lazartravica — APPROVED** [review]
@Kourin1996 Please merge after this PR: https://github.com/0xPolygon/polygon-edge/pull/334

**Pull request merged**